cetlvast/suites/unittest/test_unbounded_variant.cpp

```diff
@@ -598,36 +663,36 @@ TEST(test_unbounded_variant, assign_1_copy)
 
         ub_var dst{};
         dst = src1;
-        EXPECT_THAT(stats.ops, "@CCC~");
+        EXPECT_THAT(stats.ops, "@CC");
 
         EXPECT_THAT(get<const test&>(src1).value_, 10);
         EXPECT_THAT(get<const test&>(src1).payload_, 'X');
-        EXPECT_THAT(get<const test&>(dst).value_, 30);
+        EXPECT_THAT(get<const test&>(dst).value_, 20);
         EXPECT_THAT(get<const test&>(dst).payload_, 'X');
 
         const test value2{'Z', side_effects};
-        EXPECT_THAT(stats.ops, "@CCC~@");
+        EXPECT_THAT(stats.ops, "@CC@");
 
         const ub_var src2{value2};
-        EXPECT_THAT(stats.ops, "@CCC~@C");
+        EXPECT_THAT(stats.ops, "@CC@C");
 
         dst = src2;
-        EXPECT_THAT(stats.ops, "@CCC~@CCC~C~C~~");
+        EXPECT_THAT(stats.ops, "@CC@C~C");
 
         auto dst_ptr = &dst;
         dst          = *dst_ptr;
-        EXPECT_THAT(stats.ops, "@CCC~@CCC~C~C~~");
+        EXPECT_THAT(stats.ops, "@CC@C~C");
 
         EXPECT_THAT(get<const test&>(src2).value_, 10);
         EXPECT_THAT(get<const test&>(src2).payload_, 'Z');
-        EXPECT_THAT(get<const test&>(dst).value_, 30);
+        EXPECT_THAT(get<const test&>(dst).value_, 20);
         EXPECT_THAT(get<const test&>(dst).payload_, 'Z');
     }
     EXPECT_THAT(stats.constructs, stats.destructs);
-    EXPECT_THAT(stats.ops, "@CCC~@CCC~C~C~~~~~~~");
+    EXPECT_THAT(stats.ops, "@CC@C~C~~~~~");
```

Comment:
All these shortened `stats.ops` (f.e.
`"@CCC~@CCC~C~C~~~~~~~"` →
`"@CC@C~C~~~~~"`
) are result of eliminating of the "swap" pattern from `unbounded_variant::operator=`-s.

Reply:
BTW, elimination of the "swap" pattern is needed to properly maintain the new "valueless by exception" state. Otherwise the state will end up at some temporary swap var only, and won't be properly passed/reflected on actual target objects.